### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by mark-a**
Modernize jRuby and improve on Readme and examples

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,37 +1,66 @@
...
   task :jar => :compile do
+    puts "Bundling..."
     system 'jar cvMf target/rubyfox.jar -C out .'
   end
 
-  desc "Install jars to #{sfe_root}/../__lib__"
+  desc "Install jars to #{install_lib_dir}"
   task :install_jruby do
-    cp FileList["lib/*.jar"], "#{sfe_root}/../__lib__"
+    puts "Installing Jruby..."
```
Comment (splattael) [inline Rakefile]: It's either `JRuby` or `jruby` :sweat_smile: 

```suggestion
    puts "Installing JRuby..."
```

File: Rakefile [hunk 2]
```diff
@@ -1,37 +1,66 @@
...
+    cp FileList["lib/*.jar"], install_lib_dir
   end
 
   desc "Install rubyfox.jar"
   task :install_jar => :jar do
```
Comment (splattael) [inline Rakefile]: As the task `jar` now explicitly called in https://github.com/neopoly/rubyfox/pull/1/files#diff-52c976fc38ed2b4e3b1192f8a8e24cffR21 can we drop this dependency, then? :thinking:

File: Rakefile [hunk 3]
```diff
@@ -1,37 +1,66 @@
...
-end
+else
+  extensions_dir = File.join(sfe_root, "extensions")
+  extension_dir = File.join(extensions_dir, extension_name)
+  install_lib_dir = File.join(extensions_dir, "__lib__")
 
-desc "Compile, build and install jar"
-task :default => "rubyfox"
+  unless File.directory?(install_lib_dir)
+    FileUtils.mkdir_p(install_lib_dir)
```
Comment (splattael) [inline Rakefile]: I *think* `FileUtils` is [included by default](https://github.com/ruby/rake/blob/8d3c5492b86567d8915e0fbea8597582d1fad876/lib/rake/dsl_definition.rb#L11-L12) in `Rakefile` so we could drop `FileUtils.` everywhere.

```suggestion
    mkdir_p(install_lib_dir)
```

File: Rakefile (hunk 2 above)
Comment (mark-a) [reply Rakefile]: Good catch will remove the dependency

File: Rakefile (hunk 3 above)
Comment (mark-a) [reply Rakefile]: Good catch, thanks for the review.

**Pull request merged**